spec/src/main/asciidoc/core/spi.asciidoc

```diff
@@ -1302,3 +1430,53 @@ public interface ProcessObserverMethod<T, X> {
 If any observer method of a `ProcessObserverMethod` event throws an exception, the exception is treated as a definition error by the container.
 
 If any `ProcessObserverMethod` method is called outside of the observer method invocation, an `IllegalStateException` is thrown.
+
+[[configurators]]
+
+=== Configurators interfaces
+
+CDI 2.0 introduced the following Configurators interface:
+
+* <<annotated_type_configurator>> for `AnnotatedType` configuration
+* <<injection_point_configurator>> for `InjectionPoint` configuration
+* <<bean_attributes_configurator>> for `BeanAttributes` configuration
+* <<bean_configurator>> for `Bean` configuration
+* <<observer_method_configurator>> for `ObserverMethod` configuration
+
+The container must provide implementation for all these configurators and make them available in matching container lifecycle events as defined in <<init_events>>.
+
+[[builders]]
+
+=== Builders interfaces
+
+Builders interfaces are helpers to create instances for the following SPI elements:
+
+* `AnnotatedType` built with `javax.enterprise.inject.spi.builder.BeanAttributesBuilder`,
+* `InjectionPoint` built with `javax.enterprise.inject.spi.builder.InjectionPointBuilder`,
+* `BeanAttributes` built with `javax.enterprise.inject.spi.builder.BeanAttributesBuilder`,
+* `Bean` built with `javax.enterprise.inject.spi.builder.BeanBuilder`
+* `ObserverMethod` built with `javax.enterprise.inject.spi.builder.ObserverMethodBuilder`.
+
+Each builder uses a configurator accessible with `configure()` method:
+
+* `AnnotatedTypeBuilder` uses `AnnotatedTypeConfigurator` as defined in <<annotated_type_configurator>>
+* `InjectionPointBuilder` uses `InjectionPointConfigurator` as defined in <<injection_point_configurator>>
+* `BeanAttributesBuilder` uses `BeanAttributesConfigurator` as defined in <<bean_attributes_configurator>>
+* `BeanBuilder` uses `BeanConfigurator` as defined in <<bean_configurator>>
+* `ObserverMethodBuilder` uses `ObserverMethodConfigurator` as defined in <<observer_method_configurator>>
+
+The container must provide implementation for all these builders and make them available through `ServiceLoader` for `javax.enterprise.inject.spi.builder.BuildersProvider` interface.
+
+The class `javax.enterprise.inject.spi.builder.Builders` provides static method to obtain each builder:
+
+[source, java]
+----
+public final class Builders {
+    public static <T> AnnotatedTypeBuilder<T> annotatedType(Class<T> type) {}
+    public static <T> AnnotatedTypeBuilder<T> annotatedType(AnnotatedType<T> type) {}
```

Comment:
This method is no longer a part the Builders class and hence this line should be removed.